caps/capabilities.go

```diff
@@ -92,8 +97,14 @@ func (r *Repository) Add(cap *Capability) error {
 
 // Remove removes the capability with the provided name.
 // Removing a capability that doesn't exist silently does nothing
-func (r *Repository) Remove(name string) {
-	delete(r.caps, name)
+func (r *Repository) Remove(name string) error {
+	_, ok := r.caps[name]
+	if ok {
+		delete(r.caps, name)
+		return nil
+	}
+	return NotFound{fmt.Errorf(
```

Comment:
Errors should always be returned as a pointer, so we don't need to guess when type-asserting them (you'll pretty much always fine err.(*FooError) rather than err.(FooError)).